eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/MockableArmClientProvider.cs

```diff
@@ -56,5 +64,70 @@ private MethodProvider BuildGetResourceIdMethodForResource(ResourceClientProvide
 
             return new MethodProvider(signature, body, this);
         }
+
+        private IList<MethodProvider> BuildMethodsForExtensionResource(ResourceClientProvider resource)
+        {
+            var result = new List<MethodProvider>();
+
+            var scopeParameter = new ParameterProvider("scope", $"The scope of the resource collection to get.", typeof(ResourceIdentifier));
+            var signature = new MethodSignature(
+                $"Get{resource.Name}s",
+                $"Gets a collection of {resource.ResourceCollection!.Type:C} objects within the specified scope.",
+                MethodSignatureModifiers.Public | MethodSignatureModifiers.Virtual,
+                resource.ResourceCollection!.Type,
```

Comment:
I am not sure if this has happened, but technically, an extension resource could be a singleton.
So here we should do a if first for `if (resoource.IsSingleton) {} else {}` and the code here could be in the else branch.
We could put a return empty and a diagnostic in the singleton branch for the future to complete.

Reply:
add a TODO item will handle singleton extension resource case when we actually see it.